local/resource_local_file.go

```diff
@@ -91,8 +91,9 @@ func resourceLocalFileCreate(d *schema.ResourceData, _ interface{}) error {
 		return err
 	}
 
-	checksum := sha1.Sum([]byte(content))
-	d.SetId(hex.EncodeToString(checksum[:]))
+	filenameChecksum := sha1.Sum([]byte(destination))
+	contentChecksum := sha1.Sum([]byte(content))
+	d.SetId(hex.EncodeToString(append(filenameChecksum[:], contentChecksum[:]...)))
```

Comment:
The id isn't really meaningful for this resource type (it gets all of its required information from the other attributes) so if we're going to change it anyway, maybe more straightforward to just set it to some constant value like `"-"` to make it clearer that it's not significant, and then we can remove it altogether once a later SDK version removes the requirement for having an `id` attribute. (This requirement is now an SDK-only thing; Terraform Core just treats it the same as any other attribute.)

I'm not sure if this is important, but note that anyone that was referencing `local_file.example.id` and expecting to get a SHA1 checksum will see this as a breaking change. I think we never documented this as possible, so probably not a big deal, but probably worth mentioning in the changelog once this is merged just in case.